chevah/brink
# [#2222] Add support for versioning the python distribution.

Scope
=====

This updates the paver.sh to make use of the newly versioned python-package archives.


Changes
=======

paver.sh was updated to use the versioned python when possible... but to also fall back to the archive which is not versioned.

The paver.conf was updated to include allow explicitly defining the version to use. 

Looks like solaris10 is the only one with an older version of Python.
There is also HPUX without a version at all... but hope this will be updated soon,

How to try and test the changes
===============================

reviewers: @alibotean @brunogola 

Check that changes make sense.

@dumol let me know if you have a better idea for implemeting test_version_exists


1.paver get_python

this is used by server release process 

```
# direct get for a python which is not versioned
./paver.sh get_python hpux1131 ia64
# Direct get for python with version
./paver.sh get_python solaris11 x64
```

2.Upgrade from installed unversioned version
-------------

```
# Make sure the python looks like an old one without version
rm build-ubuntu1604-x64/lib/PYTHON_PACKAGE_VERSION
./paver.sh deps
```
